### PR TITLE
Revert "[SPARK-33995][SQL] Expose make_interval as a Scala function"

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
This reverts commit e6753c9402b5c40d9e2af662f28bd4f07a0bae17.

### Why are the changes needed?
The `make_interval` function aims to construct values of the legacy interval type `CalendarIntervalType` which will be substituted by ANSI interval types (see SPARK-27790). Since the function has not been released yet, it would be better to don't expose it via public API at all.

### Does this PR introduce _any_ user-facing change?
Should not since the `make_interval` function has not been released yet.

### How was this patch tested?
By existing test suites, and GA/jenkins builds.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @MrPowers and @cloud-fan

**Comment by HyukjinKwon** [comment]
I think it's fine though since we didn't deprecate `CalendarIntervalType`. That specific type has been there from 1.5.0.

**Review by yaooqinn — APPROVED** [review]

**Comment by MrPowers** [comment]
👍 

Thanks for creating something better @MaxGekk and thanks for the ping @dongjoon-hyun.

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -2928,31 +2928,6 @@ object functions {
...
-   * @param weeks Number of weeks
-   * @param days Number of days
-   * @param hours Number of hours
-   * @param mins Number of mins
-   * @param secs Number of secs
-   * @return A datetime interval
-   * @group datetime_funcs
-   * @since 3.2.0
-   */
-  def make_interval(
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala:2945]: do we have `make_ym_interval` and `make_dt_interval` now?

Comment (HyukjinKwon) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala:2945]: I wouldn't remove until we decide to deprecate `CalendarIntervalType`?

Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala:2945]: Is it better to remove it before a release? `@since 3.2.0`

Comment (HyukjinKwon) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala:2945]: Will we deprecate and remove `CalendarIntervalType`?

Comment (MaxGekk) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala:2945]: > do we have make_ym_interval and make_dt_interval now?

@cloud-fan As SQL functions, see https://github.com/apache/spark/pull/32645 and https://github.com/apache/spark/pull/32601 . The functions haven't been exposed to Scala/Python/R APIs yet.

> Will we deprecate and remove CalendarIntervalType?

@HyukjinKwon Yes, we will I hope. The questions is only when.

Comment (HyukjinKwon) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala:2945]: okie then i am good

**Comment by MaxGekk** [comment]
Thanks all for reviews. Merging to master.

**Pull request closed without merging**